homeassistant/components/homewizard/switch.py

```diff
@@ -124,3 +132,61 @@ async def async_turn_off(self, **kwargs: Any) -> None:
     def is_on(self) -> bool:
         """Return true if switch is on."""
         return bool(self.coordinator.data["state"].switch_lock)
+
+
+class HWEnergyEnableCloudEntity(HWEnergySwitchEntity):
+    """
+    Representation of the enable cloud configuration.
+
+    Disable cloud turns off all communication to HomeWizard Cloud.
+    At this point, the device is fully local.
+    """
+
+    _attr_name = "Enable cloud"
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _attr_entity_category = EntityCategory.CONFIG
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator, entry, "cloud_enabled")
+        self.hass = hass
+        self.entry = entry
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn switch-lock on."""
+        await self.coordinator.api.system_set(cloud_enabled=True)
+        await self.coordinator.async_refresh()
+
+        persistent_notification.async_dismiss(
+            self.hass, f"{DOMAIN}_cloud_disabled_for_{self.unique_id}"
+        )
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn switch-lock off."""
+        await self.coordinator.api.system_set(cloud_enabled=False)
+        await self.coordinator.async_refresh()
+
+        persistent_notification.async_create(
```

Comment:
This is not something we should do in Home assistant. This will add more noise to the notifications screen, causing users to ignore all notifications in HA.

I suggest the HomeWizard app creates a persistent notification with their app if they want this feature.

Reply:
Hmm okey. We will see if we can add this to the app, but we can't see the difference between "offline" and "disabled" which makes this a bit hard.

But I think that the documentation itself will be sufficient for now, we will see what the impact is at HomeWizard support.

Reply:
Is a message in the log allowed?